package/src/main/java/com/azure/cosmos/cassandra/CosmosLoadBalancingPolicy.java

```diff
@@ -0,0 +1,442 @@
+    private CopyOnWriteArrayList<Node> writeLocalDcNodes;
+
+    // endregion
+
+    // region Constructors
+
+    /**
+     * Initializes a newly created {@link CosmosLoadBalancingPolicy} object from configuration.
+     *
+     * @param driverContext an object holding the context of the current driver instance.
+     * @param profileName   name of the configuration profile to apply.
+     */
+    public CosmosLoadBalancingPolicy(final DriverContext driverContext, final String profileName) {
+
+        final DriverExecutionProfile profile = driverContext.getConfig().getProfile(profileName);
+
+        this.dnsExpiryTimeInSeconds = Option.DNS_EXPIRY_TIME.getValue(profile);
+        this.globalEndpoint = Option.GLOBAL_ENDPOINT.getValue(profile);
+        this.readDatacenter = Option.READ_DATACENTER.getValue(profile);
+        this.writeDatacenter = Option.WRITE_DATACENTER.getValue(profile);
+
+        this.validate();
+    }
+
+    // endregion
+
+    // region Methods
+
+    /**
+     * Closes the current {@link CosmosLoadBalancingPolicy} instance.
+     */
+    @Override
+    public void close() {
+        // nothing to do
+    }
+
+    /**
+     * Initializes the list of hosts in read, write, local, and remote categories.
+     *
+     * @param nodes the nodes to be examined.
+     * @param distanceReporter an object that the policy uses to signal decisions it makes about node distances.
+     */
+    @Override
+    public void init(@NonNull final Map<UUID, Node> nodes, @NonNull final DistanceReporter distanceReporter) {
+
+        List<InetAddress> dnsLookupAddresses = new ArrayList<>();
+
+        if (!this.globalEndpoint.isEmpty()) {
+            dnsLookupAddresses = Arrays.asList(this.getLocalAddresses());
+        }
+
+        this.readLocalDcNodes = new CopyOnWriteArrayList<>();
+        this.writeLocalDcNodes = new CopyOnWriteArrayList<>();
+        this.remoteDcNodes = new CopyOnWriteArrayList<>();
+
+        for (final Node node : nodes.values()) {
+
+            final String datacenter = node.getDatacenter();
+            NodeDistance distance = NodeDistance.IGNORED;
+
+            if (!this.readDatacenter.isEmpty() && Objects.equals(datacenter, this.readDatacenter)) {
+                this.readLocalDcNodes.add(node);
+                distance = NodeDistance.LOCAL;
+            }
+
+            if ((!this.writeDatacenter.isEmpty() && Objects.equals(datacenter, this.writeDatacenter))
+                || dnsLookupAddresses.contains(this.getAddress(node))) {
+                this.writeLocalDcNodes.add(node);
+                distance = NodeDistance.LOCAL;
+            } else {
+                assert distance == NodeDistance.IGNORED;
+                this.remoteDcNodes.add(node);
+                distance = NodeDistance.REMOTE;
+            }
+
+            distanceReporter.setDistance(node, distance);
+        }
+
+        this.index.set(new Random().nextInt(Math.max(nodes.size(), 1)));
```

Comment:
What does this do?

Reply:
The call to `this.index.set` selects a new random start in the search for a node search. It's directly ported from `java-driver-3`.